src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationHelper.cs

```diff
@@ -0,0 +1,138 @@
+            }
+
+            if (jsonSpan.IsEmpty)
+            {
+                return default; // Return default JsonElement if the file is empty.
+            }
+
+            var reader = new Utf8JsonReader(jsonSpan, isFinalBlock: true, state: default);
+            using var doc = JsonDocument.ParseValue(ref reader);
+
+            return doc.RootElement.Clone();
+        }
+
+        private static List<WorkerDescriptionProfile> ReadWorkerDescriptionProfiles(JsonElement profilesElement,
+                                                                            JsonSerializerOptions jsonSerializerOptions,
+                                                                            IWorkerProfileManager profileManager,
+                                                                            ILogger logger)
+        {
+            var profiles = profilesElement.Deserialize<IList<WorkerProfileDescriptor>>(jsonSerializerOptions);
+
+            if (profiles == null || profiles.Count <= 0)
+            {
+                return new List<WorkerDescriptionProfile>(0);
+            }
+
+            var descriptionProfiles = new List<WorkerDescriptionProfile>(profiles.Count);
+
+            try
+            {
+                foreach (var profile in profiles)
+                {
+                    var profileConditions = new List<IWorkerProfileCondition>(profile.Conditions.Count);
+
+                    foreach (var descriptor in profile.Conditions)
+                    {
+                        if (!profileManager.TryCreateWorkerProfileCondition(descriptor, out IWorkerProfileCondition condition))
+                        {
+                            // Failed to resolve condition. This profile will be disabled using a mock false condition
+                            logger.LogInformation("Profile {name} is disabled. Cannot resolve the profile condition {condition}", profile.ProfileName, descriptor.Type);
+                            condition = new FalseCondition();
+                        }
+
+                        profileConditions.Add(condition);
+                    }
+
+                    descriptionProfiles.Add(new(profile.ProfileName, profileConditions, profile.Description));
+                }
+            }
+            catch (Exception)
+            {
+                throw new FormatException("Failed to parse profiles in worker config.");
+            }
+
+            return descriptionProfiles;
+        }
+
+        private static void GetWorkerDescriptionFromAppSettings(RpcWorkerDescription workerDescription, Dictionary<string, string> languageWorkersSettings)
+        {
+            if (languageWorkersSettings.TryGetValue($"{RpcWorkerConstants.LanguageWorkersSectionName}:{workerDescription.Language}:{WorkerConstants.WorkerDescriptionDefaultExecutablePath}", out string defaultExecutablePathSetting) && defaultExecutablePathSetting is not null)
```

Comment:
If we parse LanguageWorkerSettings elsewhere like i suggested, this gets simpler.